projects/plugins/boost/app/admin/class-regenerate-admin-notice.php

```diff
@@ -21,54 +22,78 @@ public static function enable() {
 	}
 
 	public static function dismiss() {
-		Transient::delete( 'regenerate_admin_notice', false );
+		Transient::delete( 'regenerate_admin_notice' );
 	}
 
 	public static function is_enabled() {
 		return Transient::get( 'regenerate_admin_notice', false );
 	}
 
+	public static function enable_suggestion() {
+		Transient::set( 'regenerate_admin_suggestion', true );
+	}
+
+	public static function dismiss_suggestion() {
+		Transient::delete( 'regenerate_admin_suggestion' );
+	}
+
+	public static function is_suggestion_enabled() {
+		return Transient::get( 'regenerate_admin_suggestion', false );
+	}
+
 	/**
 	 * Helper method to generate a dismissal link for this message.
 	 */
 	private static function get_dismiss_url() {
 		return add_query_arg(
 			array(
 				self::$dismissal_key => '',
+				'nonce'              => wp_create_nonce( 'jb_dismiss_notice' ),
 			)
 		);
 	}
 
 	public static function maybe_handle_dismissal() {
-		// We're okay dismissing the notice without nonce verification.
-		// phpcs:disable WordPress.Security.NonceVerification.Recommended
-		if ( ! is_admin() || ! current_user_can( 'manage_options' ) || ! isset( $_GET[ self::$dismissal_key ] ) ) {
+		if ( ! is_admin()
+			|| ! current_user_can( 'manage_options' )
+			|| ! isset( $_GET[ self::$dismissal_key ], $_GET['nonce'] )
+			|| ! wp_verify_nonce( sanitize_text_field( wp_unslash( $_GET['nonce'] ) ), 'jb_dismiss_notice' )
+		) {
 			return;
 		}
+
+		// Mark the critical CSS as fresh so we don't show the suggestion again in a period of time.
+		Critical_CSS_State::set_fresh( true );
```

Comment:
I like this design. It allows users to turn off notices for a while, but they will eventually return to let users know they need to update. That way, users won't dismiss them and forget forever, letting their Critical CSS go stale.

I wonder if we need a "never show me this again" option too?

Reply:
"Never show me this again" would be an excellent option. Besides improving user experience, we can also notify the user about the importance of keeping C.CSS updated if they click it. We can also show us our premium option and give them a chance to upgrade.